### PR TITLE
Revert "Do lowering of java_new as a function-level pass"

### PR DESCRIPTION
**Pull request by smowton**
This reverts commit 166563fe74426954f9bc860f62e1e3babc3450de.

remove_java_new was introduced because ID_java_new represented an abstract,
unsized allocation whereas ID_allocate carried a definite size, and running
java_bytecode_convert_method on a function could change a type's size by
discovering previously-unknown fields belonging to opaque types.

However, as of e86e2a026b4ba9bb89f875c816ab5e534d82b8fb the sizes of types are
ascertained earlier (during java_bytecode_languaget::typecheck), and so there
is now no problem with using ID_allocate from the outset.

### CODE REVIEW

**Comment by smowton** [comment]
Test-gen bump: https://github.com/diffblue/test-gen/pull/1464

File: src/goto-programs/class_identifier.h [hunk 1]
```diff
@@ -15,16 +15,10 @@ Author: Chris Smowton, chris.smowton@diffblue.com
...
 exprt get_class_identifier_field(
   const exprt &this_expr,
   const symbol_typet &suggested_type,
   const namespacet &ns);
 
-void set_class_identifier(
-  struct_exprt &expr,
-  const namespacet &ns,
-  const symbol_typet &class_type);
-
```
Comment (tautschnig) [inline src/goto-programs/class_identifier.h]: While I appreciate the clean revert: isn't this actually a desirable place for `set_class_identifier` to be declared? (And, in line with that, the implementation in the .cpp file?)

**Comment by smowton** [comment]
@tautschnig amended as suggested

**Review by tautschnig — APPROVED** [review]

File: src/goto-programs/goto_convert.cpp [hunk 2]
```diff
@@ -766,7 +766,10 @@ void goto_convertt::convert_assign(
   else if(rhs.id()==ID_side_effect &&
           rhs.get(ID_statement)==ID_java_new)
   {
-    copy(code, ASSIGN, dest);
+    Forall_operands(it, rhs)
```
Comment (thk123) [inline src/goto-programs/goto_convert.cpp:769]: Perhaps we can use range based for loop here?

**Comment by smowton** [comment]
@peterschrammel could you review?

**Comment by smowton** [comment]
Or @chrisr-diffblue ?

File: src/goto-programs/goto_convert.cpp (hunk 2 above)
Comment (smowton) [reply src/goto-programs/goto_convert.cpp:769]: Going to stick with a plain revert there

**Review by mgudemann — APPROVED** [review]
LGTM

**Review by chrisr-diffblue — APPROVED** [review]
Looks ok to me too.

**Pull request merged**